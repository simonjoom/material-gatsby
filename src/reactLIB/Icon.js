import React from 'react';
import PropTypes from 'prop-types';
import constants from './constants';
import cx from 'classnames';

class Icon extends React.Component {
  static defaultProps = {
    type: 'awesome'
  };
  render() {
    const awesome = this.props.type == 'awesome';
    const classes = {
      fa: awesome,
      'material-icons': !awesome
    };
    constants.PLACEMENTS.forEach(p => {
      classes[p] = this.props[p];
    });
    constants.ICON_SIZES.forEach(s => {
      classes[s] = this.props[s];
    });
    return (
      <i
        style={this.props.style}
        className={cx(classes, awesome ? 'fa-' + this.props.className : '')}
      >
        {!awesome && this.props.className}
      </i>
    );
  }
}

Icon.propTypes = {
  type: PropTypes.string,
  /*
   * Classname passed to i tag
   */
  className: PropTypes.string,
  /*
   * Icon type: <a href='https://material.io/icons/'>https://material.io/icons/</a>
   */
  children: PropTypes.string,
  /*
   * Placement for icon if used beside a text ↓
   */
  left: PropTypes.bool,
  center: PropTypes.bool,
  right: PropTypes.bool,
  /*
   * Sizes for icons ↓
   */
  tiny: PropTypes.bool,
  small: PropTypes.bool,
  medium: PropTypes.bool,
  large: PropTypes.bool
};

export default Icon;
