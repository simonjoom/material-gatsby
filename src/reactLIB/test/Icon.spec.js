import React from 'react';
import { shallow } from 'enzyme';
import Icon from '../src/Icon';

describe('<Icon />', () => {
  let wrapper = shallow(<Icon>cloud</Icon>);

  test('renders an icon', () => {
    wrapper = shallow(<Icon>cloud</Icon>);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts size as a prop', () => {
    wrapper = shallow(<Icon large>cloud</Icon>);
    expect(wrapper).toMatchSnapshot();
  });
  test('accepts placement as a prop', () => {
    wrapper = shallow(<Icon left>cloud</Icon>);
    expect(wrapper).toMatchSnapshot();
  });
});
