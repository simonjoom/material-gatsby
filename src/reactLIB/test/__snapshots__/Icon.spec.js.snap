// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> accepts placement as a prop 1`] = `
<i
  className="material-icons left"
>
  cloud
</i>
`;

exports[`<Icon /> accepts size as a prop 1`] = `
<i
  className="material-icons large"
>
  cloud
</i>
`;

exports[`<Icon /> renders an icon 1`] = `
<i
  className="material-icons"
>
  cloud
</i>
`;
